agent/src/main/java/io/atomix/agent/AtomixAgent.java

```diff
@@ -78,13 +76,13 @@ public static void main(String[] args) throws Exception {
         .help("The Atomix configuration file");
     parser.addArgument("--core-nodes", "-n")
         .nargs("*")
-        .type(nodeArgumentType)
+        .type(nodeArgumentType.apply(Node.Type.CORE))
         .metavar("NAME:HOST:PORT")
         .required(false)
         .help("Sets the core cluster configuration");
     parser.addArgument("--bootstrap-nodes", "-b")
         .nargs("*")
-        .type(nodeArgumentType)
+        .type(nodeArgumentType.apply(Node.Type.DATA))
```

Comment:
data for bootstrap nodes? I haven't used args4j much..

Reply:
If `--core-nodes` are provided, they'll be used as bootstrap nodes, so the assumption is that any `--bootstrap-nodes` provided will be `DATA` nodes

Reply:
what if they aren't? is it corrected with gossip?

Reply:
if `--core-nodes` aren't provided then presumably there _are_ no core nodes. This is an unavoidable limitation of consensus. A static consensus configuration must be provided or correctness is risked.

Reply:
I mean what if I start a node as core and configure that node as data in another node.

Reply:
The problem is, if different nodes are started with different sets of `CORE` nodes, they'll count votes differently and can thus elect multiple leaders, leading to split brain. The CORE configuration must be the same on all nodes at startup. There's no way around that limitation. A single source must be assumed - either in a provided configuration or by querying a single external source.